File: perl/e11.branch.pl
#!/usr/bin/perl -w
use v5.14;
  
print "Input a character: "; my $keypress=getc(STDIN);
  
for ($keypress) {
  print "Lowercase letter\n" when /[[:lower:]]/;
  print "Uppercase letter\n" when /[[:upper:]]/;
  print "Digit\n" when /[[:digit:]]/;
  default { print "Punctuation, whitespace, or other\n"; }
}

File: php/e11.branch.php
#!/usr/bin/php
<?php
echo "Input a character: "; $Keypress = fgetc(STDIN);
 
switch (TRUE) {
  case preg_match("/[[:lower:]]/", $Keypress):
    echo "Lowercase letter\n";
    break;
  case preg_match("/[[:upper:]]/", $Keypress):
    echo "Uppercase letter\n";
    break;
  case preg_match("/[[:digit:]]/", $Keypress):
    echo "Digit\n";
    break;
  default:
    echo "Puncuation, whitespace, or other\n";
}
?>
